CONTRIBUTING.md

```diff
@@ -0,0 +1,223 @@
+[glance-operator design decisions document](docs/dev/design-decisions.md).
+
+### How to contribute?
+
+This project is [using pull
+request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests)
+to submit changes, but it is not currently using the [issues
+section](https://github.com/openstack-k8s-operators/glance-operator/issues) or
+any other GitHub feature to track the project's bugs or features and instead it
+is expected that whoever finds an issue will fix it or will find someone else
+to work on it.
+
+### Testing changes
+
+Developers submitting PRs are expected to have run the code and manually
+verified the functionality before submitting the PR, with the exception
+of documentation only and CI only PRs.
+
+There are multiple ways to test code changes, but in general terms they can be
+split into 2 categories: running things locally or in a container in the
+OpenShift cluster.
+
+Running things locally is considerably faster but it doesn't use the real ACLs
+(RBACs) the glance-operator would use on a normal deployment. On the other hand
+running things in OpenShift is slower (because we have to build a new
+container) but runs as close as a real deployment as we can.
+
+Each of the approaches has its own advantages and disadvantages, so different
+variants of both approaches will be covered for readers to choose the one they
+prefer.
+
+The following list of articles assumes that podified OpenStack has been
+deployed using the openstack-operator as described in the [Getting started
+section](README.md#getting-started):
+
+- [Run operator locally](docs/dev/local.md)
+
+### Debugging
+
+When working on the cinder-operator there will be times where things won't work
+as expected and we'll need to debug things.
+
+In the [debugging article](docs/dev/debug.md) we present some ideas to help you
+figure things out.
+
+### Pull Requests
+
+While the pull request flow is used for submitting new issues and for
+reviewing, the git repository should **always** be the source of truth, so all
+decisions made through the reviewing and design phases should be properly
+reflected in the final code, code comments, documentation, and git commit
+message that is merged.  It should not be necessary to go to a PR in GitHub and
+go through the comments to know what a commit is meant to do or why it is doing
+it.
+
+*One-liner* commit messages should be avoided like the plague.  Please refer to
+the [commit messages guide line](#commit-messages) for good practices on commit
+messages.
+
+The glance-operator project will not squash all pull request commits into a
+single commit but will look to preserve all submitted individual commits
+instead using a merge strategy instead.  This means that we can have both
+single commit and as multi-commit PRs, and both have their places. It's all
+about how and when to split changes.
+
+#### Dependency Management
```

Comment:
I think we don't have this today, we can change this to say which is good practice state in the `commitmsg` the dependencies, but there's no automation to put things together like we used to do with zuul.
What do you think about changing this part and just leave it as good practice to follow?
In case we have some automation in the future we can always consider a follow up.

Reply:
Sounds good!